client/postit/src/components/SignUp.js

```diff
@@ -0,0 +1,123 @@
+import "bootstrap/dist/css/bootstrap.css";
+import "../css/custom.css";
+import { useState, useEffect } from "react";
+function SignUp({ onRegistration, onError }) {
+  const initialValues = {
+    username: "",
+    email: "",
+    password: "",
+  };
+  const [values, setValues] = useState(initialValues);
+  const [errMsg, setErrMsg] = useState("");
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+
+    setValues({ ...values, [name]: value });
+  };
+
+  useEffect(() => {
+    setErrMsg("");
+  }, [values.username, values.email, values.password]);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const result = await onRegistration(
+      values.username,
+      values.password,
+      values.email
+    );
+
+    if (result === 404) {
```

Comment:
This was abit confusing for me at first site, so if the return values are changed as suggested above in the onRegistration method , then here we would have something like 
`if (result.message !== 'success'){
   if(result.status === 400){ \\do something}
   \\ and so on
}`
Looks a bit like more code, but is more readable